File: frontend/demo/component/board/board-nested.ts
import 'Frontend/demo/init'; // hidden-source-line

import '@vaadin/vaadin-board/vaadin-board';
import { html, LitElement } from 'lit';
import { customElement } from 'lit/decorators.js';
import { applyTheme } from 'Frontend/generated/theme';
import './example-indicator';
import './example-statistics';

@customElement('board-nested')
export class Example extends LitElement {
  constructor() {
    super();
    this.classList.add('board-nested');
  }

  protected createRenderRoot() {
    const root = super.createRenderRoot();
    // Apply custom theme (only supported if your app uses one)
    applyTheme(root);
    return root;
  }

  render() {
    return html`
      <!-- tag::snippet[] -->
      <vaadin-board>
        <vaadin-board-row>
          <example-statistics board-cols="2"></example-statistics>
          <vaadin-board-row board-cols="1">
            <example-indicator
              current="745"
              change="+33.7"
              title="Current users"
            ></example-indicator>
            <example-indicator
              current="18%"
              change="+3.9"
              title="Conversion rate"
            ></example-indicator>
          </vaadin-board-row>
        </vaadin-board-row>
      </vaadin-board>
      <!-- end::snippet[] -->
    `;
  }
}

File: src/main/java/com/vaadin/demo/component/board/BoardNested.java
package com.vaadin.demo.component.board;

import com.vaadin.flow.component.board.Board;
import com.vaadin.flow.component.board.Row;
import com.vaadin.flow.component.html.Div;
import com.vaadin.flow.router.Route;
import com.vaadin.demo.DemoExporter; // hidden-source-line

@Route("board-nested")
public class BoardNested extends Div {
    public BoardNested() {
        // tag::snippet[]
        Row rootRow = new Row();
        rootRow.add(new ExampleStatistics(), 2);

        Row nestedRow = new Row(
                new ExampleIndicator("Current users", "745", "+33.7"),
                new ExampleIndicator("Conversion rate", "18%", "+3.9")
        );
        rootRow.addNestedRow(nestedRow);

        Board board = new Board();
        board.add(rootRow);
        // end::snippet[]

        add(board);
        this.setClassName("board-nested");
    }

    public static class Exporter extends DemoExporter<BoardNested> { // hidden-source-line
    } // hidden-source-line
}
